OpenRA.Mods.Common/Traits/GivesExperience.cs

```diff
@@ -54,9 +55,15 @@ public void Killed(Actor self, AttackInfo e)
 				? info.Experience
 				: valued != null ? valued.Cost : 0;
 
+			Util.ApplyPercentageModifiers(exp, self.TraitsImplementing<IGivesExperienceModifier>().Select(x => x.GetGivesExperienceModifier()));
+
 			var killer = e.Attacker.TraitOrDefault<GainsExperience>();
 			if (killer != null)
-				killer.GiveExperience(Util.ApplyPercentageModifiers(exp, new[] { info.ActorExperienceModifier }));
+			{
+				var killerexpmodifier = new[] { info.ActorExperienceModifier }
```

Comment:
Style nit: camelCase for variable names.

The LINQ can also be simplified a bit:
```csharp
var killerExperienceModifier = e.Attacker.TraitsImplementing<IGainsExperienceModifier>()
	.Select(x => x.GetGainsExperienceModifier())
	.Append(info.ActorExperienceModifier);

```